### PR TITLE
Clean up of HHAnalyzer, more GenAnalyzer methods, add GenJet/GenParton info to Jet

### PR DESCRIPTION
**Pull request by pablodecm**
Main changes affect HHAnalyzer but some modifications have been carried out over the GenAnalyzer to include methods which allow getting GenLevel info when two Higgs boson are present.

### CODE REVIEW

File: plugins/ObjectsFormat.cc [hunk 1]
```diff
@@ -283,6 +283,17 @@ void ObjectsFormat::ResetJetType(JetType& I) {
...
+    I.phiGenJ     = -4.;
+    I.massGenJ    = -10.;
+    I.ptGen       = -10.;
+    I.etaGen      = -4.;
+    I.phiGen      = -4.;
+    I.massGen     = -10.;
+    I.ptLhe       = -10.;
+    I.etaLhe      = -4.;
+    I.phiLhe      = -4.;
     I.CSV         = -99.;
```
Comment (jpazzini) [inline plugins/ObjectsFormat.cc:297]: I don't see the inclusion of these variables in the JETtype struct (L325).

File: plugins/Objects.h [hunk 2]
```diff
@@ -85,6 +85,17 @@ struct JetType {
...
+    float etaGenJ = -4.;
+    float phiGenJ = -4.;
+    float massGenJ = -10.;
+    float ptGen = -10.;
+    float etaGen = -4.;
+    float phiGen = -4.;
+    float massGen = -10.;
+    float ptLhe = -10.;
+    float etaLhe = -4.;
+    float phiLhe = -4.;
```
Comment (jpazzini) [inline plugins/Objects.h:98]: From line 88 to 98, remove the initialization values, and place them in line 77 together with the other ones.

File: plugins/ObjectsFormat.cc [hunk 3]
```diff
@@ -261,15 +261,15 @@ void ObjectsFormat::FillJetType(JetType& I, const pat::Jet* R, bool isMC) {
...
     I.isTight     = R->hasUserInt("isTight") ? R->userInt("isTight") : false;
     I.isTightLepVeto     = R->hasUserInt("isTightLepVeto") ? R->userInt("isTightLepVeto") : false;
-    I.isCSVL      = false;//IsBTagged(Tagger, 1, I.CSV);
-    I.isCSVM      = false;//IsBTagged(Tagger, 2, I.CSV);
-    I.isCSVT      = false;//IsBTagged(Tagger, 3, I.CSV);
-    I.isMatched   = (I.mother==25);
+    I.isCSVL      = R->bDiscriminator("pfCombinedInclusiveSecondaryVertexV2BJetTags")>0.460 ? true : false;
+    I.isCSVM      = R->bDiscriminator("pfCombinedInclusiveSecondaryVertexV2BJetTags")>0.800 ? true : false;
+    I.isCSVT      = R->bDiscriminator("pfCombinedInclusiveSecondaryVertexV2BJetTags")>0.935 ? true : false;
 }
```
Comment (jpazzini) [inline plugins/ObjectsFormat.cc:273]: Please be aware that these are the discriminators out of the box, with no reweighing (or reshaping) applied yet to account for the b-tagging SF.

File: plugins/ObjectsFormat.cc [hunk 4]
```diff
@@ -283,6 +283,17 @@ void ObjectsFormat::ResetJetType(JetType& I) {
...
+    I.etaGenJ     = -4.;
+    I.phiGenJ     = -4.;
+    I.massGenJ    = -10.;
+    I.ptGen       = -10.;
+    I.etaGen      = -4.;
+    I.phiGen      = -4.;
+    I.massGen     = -10.;
+    I.ptLhe       = -10.;
+    I.etaLhe      = -4.;
+    I.phiLhe      = -4.;
```
Comment (jpazzini) [inline plugins/ObjectsFormat.cc:296]: I don't see the inclusion of these variables (in their correct order) into the JetType struct [see line 325]. Please add.

**Review by jpazzini — CHANGES_REQUESTED** [review]
Looks fine. Just a couple of comments, quick to implement.
This PR provide some variation involving the Jet type, so please test carefully that all the variables defined in the Jet structs are still consistent.

File: plugins/Objects.h (hunk 2 above)
Comment (pablodecm) [reply plugins/Objects.h:98]: in-class member initialization is equivalent for the compiler and much easier to read in my opinion

**Review by jpazzini — CHANGES_REQUESTED** [review]
From what I see here, the issue with the GenAnalyzer is that you started from an old version of the code, not yet including the latest mods from Lisa.

You should be able to fix it by getting the latest version of it and resubmitting your changes...

**Comment by pablodecm** [comment]
there is the conflict with the GenAnalyzer which I cannot fix because I do not have write access, will you take care of fixing it?

File: plugins/Objects.h (hunk 2 above)
Comment (jpazzini) [reply plugins/Objects.h:98]: I might agree with you, but let's just use one style for the entire file.

**Comment by jpazzini** [comment]
OK, no problem. I can take care of it... but you wouldn't have it in the first place if you started from the latest PR, I think.

File: plugins/Objects.h (hunk 2 above)
Comment (pablodecm) [reply plugins/Objects.h:98]: ok, I will push now an update using old-school initialization list then

File: plugins/ObjectsFormat.cc (hunk 3 above)
Comment (MDallOsso) [reply plugins/ObjectsFormat.cc:273]: It the discriminator used to fill .CSV so I suppose it is the 'out of the box'.
Do you use the reshaped one in the analysis?

**Comment by pablodecm** [comment]
yes, that is right but Lisa's pull request was not merged when we started with this set of modifications, indeed I think we have to modify our python configuration file also to include this pythiaLO thingy

File: plugins/ObjectsFormat.cc (hunk 3 above)
Comment (jpazzini) [reply plugins/ObjectsFormat.cc:273]: In the VZ(lep) analysis we actually don't use b-tagging at all. Lisa is starting to have a look at the b-tagging for the 0-leptons VZ analysis but so far I think she's still used the CSV discriminator out of the box with no SF applied.

**Comment by jpazzini** [comment]
Ok. Nevermind then... It should be fine now.

**Pull request merged**